plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/IngestActionFilter.java

```diff
@@ -137,4 +142,106 @@ public void failed(Exception e) {
     public int order() {
         return Integer.MAX_VALUE;
     }
+
+    final static class BulkProcessingState implements Iterator<ActionRequest> {
```

Comment:
I renamed it to `BulkRequestSource`, because right now it is just a source of bulk request items and it helps creating the modified bulk request if needed.